backend/src/models/department.entity.ts

```diff
@@ -10,13 +10,19 @@ export class Department {
     name: string
 
     @OneToOne(() => Position)
-    @JoinColumn()
+    @JoinColumn({ name: "departmentHeadId" })
     departmentHead: Position
 
+    @Column({ type: "int", nullable: true })
+    departmentHeadId: number
```

Comment:
It seems like it might be possible to add the JoinColumns without adding these extra fields according to this [post](https://stackoverflow.com/questions/51537780/typeorm-joining-when-we-have-one-to-many-and-many-to-one-relationship) by adding a `referencedColumnName`. If we're able to do that, it would clean up the model a little

Reply:
From what I saw, I think we still have to explicitly have the column so that we can save a new entity using foreign ids instead of the entire foreign entities. Not 100% certain until we're able to test with the service, but we haven't finished that yet

Reply:
Can you point me to where you've seen that? Also, can we try without adding these columns first, and then add them later if we find that it's necessary? So just to clarify, I mean do this for all joinColumns:

```
@JoinColumn({ name: "departmentHeadId", referencedColumnName: "id" })
departmentHead: Position
```